src/main/java/hudson/plugins/sshslaves/SFTPFileSystem.java

```diff
@@ -38,9 +39,9 @@
  * @author Kohsuke Kawaguchi
  */
 class SFTPFileSystem implements FileSystem {
-    private final SFTPClient sftp;
+    private final SFTPv3Client sftp;
 
-    public SFTPFileSystem(SFTPClient sftp) {
+    public SFTPFileSystem(SFTPv3Client sftp) {
```

Comment:
Binary incompatible. Maybe makes sense to create a new SFTPv3FileSystem class instead